File: VantaSpeech/Core/Storage/Recording.swift
import Foundation
import SwiftData

@Model
final class Recording {
    var id: UUID
    var title: String
    var createdAt: Date
    var duration: TimeInterval
    var audioFileURL: String
    var transcriptionText: String?
    var summaryText: String?
    var isTranscribed: Bool
    var isUploading: Bool

    init(
        id: UUID = UUID(),
        title: String,
        createdAt: Date = Date(),
        duration: TimeInterval = 0,
        audioFileURL: String,
        transcriptionText: String? = nil,
        summaryText: String? = nil,
        isTranscribed: Bool = false,
        isUploading: Bool = false
    ) {
        self.id = id
        self.title = title
        self.createdAt = createdAt
        self.duration = duration
        self.audioFileURL = audioFileURL
        self.transcriptionText = transcriptionText
        self.summaryText = summaryText
        self.isTranscribed = isTranscribed
        self.isUploading = isUploading
    }
}

extension Recording {
    var formattedDuration: String {
        let minutes = Int(duration) / 60
        let seconds = Int(duration) % 60
        return String(format: "%02d:%02d", minutes, seconds)
    }

    var formattedDate: String {
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        formatter.timeStyle = .short
        return formatter.string(from: createdAt)
    }
}

File: Vanta Sppech Android/app/src/main/java/com/vanta/speech/core/domain/model/Recording.kt
package com.vanta.speech.core.domain.model

import java.time.Duration
import java.time.Instant
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.util.Locale
import java.util.UUID

data class Recording(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val createdAt: Instant = Instant.now(),
    val duration: Duration = Duration.ZERO,
    val audioFilePath: String,
    val transcriptionText: String? = null,
    val summaryText: String? = null,
    val isTranscribed: Boolean = false,
    val isUploading: Boolean = false,
    val isSummaryGenerating: Boolean = false,
    val preset: RecordingPreset? = null
) {
    val formattedDuration: String
        get() {
            val hours = duration.toHours()
            val minutes = duration.toMinutes() % 60
            val seconds = duration.seconds % 60
            return if (hours > 0) {
                String.format(Locale.US, "%02d:%02d:%02d", hours, minutes, seconds)
            } else {
                String.format(Locale.US, "%02d:%02d", minutes, seconds)
            }
        }

    val formattedDate: String
        get() {
            val formatter = DateTimeFormatter.ofPattern("d MMMM, HH:mm", Locale("ru"))
            return createdAt.atZone(ZoneId.systemDefault()).format(formatter)
        }

    val formattedShortDate: String
        get() {
            val formatter = DateTimeFormatter.ofPattern("d MMM", Locale("ru"))
            return createdAt.atZone(ZoneId.systemDefault()).format(formatter)
        }
}
